src/main/java/com/arpnetworking/kairosdb/aggregators/HistogramFilterAggregator.java

```diff
@@ -0,0 +1,281 @@
+            } else {
+                lowerBound = truncate(value);
+                upperBound = binInclusiveBound(value);
+            }
+
+            //=================================================================
+            /*
+             * TODO(Joey Jackson):
+             *  DESC. The refactored code below uses the .compare method added to the
+             *  FilterOp Enum in the base KairosDB library. The pull request listed below
+             *  must be merged before this cleaner version of the code can be used. Until then,
+             *  the code below is incompatible with the base KairosDB library, but is compatible
+             *  with the fork by ddimensia.
+             *  Ref:
+             *      https://github.com/kairosdb/kairosdb/pull/555
+             *      https://github.com/ddimensia/kairosdb
+            if (_filterop == FilterAggregator.FilterOperation.EQUAL) {
+                if (_filterinc == FilterIndeterminate.DISCARD) {
+                    return _threshold >= lowerBound && _threshold <= upperBound;
+                } else if (_filterinc == FilterIndeterminate.KEEP) {
+                    return false;
+                }
+            } else {
+                final boolean thresholdAcceptsLowerBound = _filterop.compare(lowerBound, _threshold);
+                final boolean thresholdAcceptsUpperBound = _filterop.compare(upperBound, _threshold);
+                return _filterinc.shouldDiscard(thresholdAcceptsLowerBound, thresholdAcceptsUpperBound);
+            }
+            */
+            //=================================================================
+            final boolean thresholdAcceptsLowerBound;
+            final boolean thresholdAcceptsUpperBound;
+            switch (_filterop) {
+                case LTE:
+                    thresholdAcceptsLowerBound = lowerBound <= _threshold;
+                    thresholdAcceptsUpperBound = upperBound <= _threshold;
+                    return _filterinc.shouldDiscard(thresholdAcceptsLowerBound, thresholdAcceptsUpperBound);
+                case LT:
+                    thresholdAcceptsLowerBound = lowerBound < _threshold;
+                    thresholdAcceptsUpperBound = upperBound < _threshold;
+                    return _filterinc.shouldDiscard(thresholdAcceptsLowerBound, thresholdAcceptsUpperBound);
+                case GTE:
+                    thresholdAcceptsLowerBound = lowerBound >= _threshold;
+                    thresholdAcceptsUpperBound = upperBound >= _threshold;
+                    return _filterinc.shouldDiscard(thresholdAcceptsLowerBound, thresholdAcceptsUpperBound);
+                case GT:
+                    thresholdAcceptsLowerBound = lowerBound > _threshold;
+                    thresholdAcceptsUpperBound = upperBound > _threshold;
+                    return _filterinc.shouldDiscard(thresholdAcceptsLowerBound, thresholdAcceptsUpperBound);
+                default: // EQUAL
```

Comment:
I would prefer to call out the case for `EQUAL` and then have default throw. Effectively, this means that adding a new enum value would reveal the requirement to add a branch here. This is also why the enum method pattern is superior because it requires you to define that behavior on the enum when adding a new element.